add kaldi to speech rec launch file

MatthijsBurgh opened the pull request.

MatthijsBurgh committed:
add kaldi to speech rec launch file

MatthijsBurgh commented:
@ar13pit ping ;)

ar13pit commented on launch/interaction/speech_recognition.launch:
```diff
@@ -4,6 +4,7 @@
 	<!-- Get the machine file -->
 	<arg name="machine" default="localhost"/>
         <arg name="dragonfly" default="true" />
+        <arg name="kaldi" default="false" />
```
A couple of lines are missing:
```
    <!-- Kaldi model path default to ~/data/kaldi_speech_recognition -->
    <arg name="model_path" default="$(env HOME)/data/speech_models/model/"/>
    <param name="/kaldi_model_path" type="string" value="$(arg model_path)"/>
```

MatthijsBurgh committed:
add missing arg & param of kaldi

MatthijsBurgh commented on launch/interaction/speech_recognition.launch:
```diff
@@ -3,7 +3,15 @@
+    <group if="$(arg kaldi)">
+        <!-- Kaldi model path default to ~/data/kaldi_speech_recognition -->
+        <arg name="model_path" default="$(env HOME)/data/speech_models/model/"/>
+        <param name="/kaldi_model_path" type="string" value="$(arg model_path)"/>
```
@ar13pit why the absolute param path? This shouldn't be the case in my opinion

ar13pit replied:
In the `install.bash` file of the target, I download the trained model into this directory that's why I put it up this way.

If you have a better way of using this parameter then lets use that

MatthijsBurgh replied:
No it is about the param **name**.  This is not a param which is used by multiple nodes. So why is it a global param? I would like to see it as an private parameter.

ar13pit replied:
Definitely. It needs to be private and not global.

MatthijsBurgh replied:
@ar13pit Do you apply the change? This maybe also requires a change in the node

ar13pit replied:
You make the appropriate change here and I'll fix the node after that

MatthijsBurgh replied:
Done

MatthijsBurgh committed:
kaldi model path as private param

param will be hmi/hmi_kaldi_node/kaldi_model_path. Can be access inside the node by ~kaldi_model_path.
